metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -918,3 +927,182 @@ def test_capability(self, project_id: str) -> None:
     def report_status(self, step: str, status: bool) -> None:
         if self.redundant_run_skip_handler:
             self.redundant_run_skip_handler.report_current_run_status(step, status)
+
+    def gen_lineage_workunits_for_external_table(
+        self,
+        dataset_urn: str,
+        ddl: Optional[str],
+        graph: Optional[DataHubGraph] = None,
+    ) -> Iterable[MetadataWorkUnit]:
+
+        if not ddl:
+            return
+
+        # Expect URIs in `uris=[""]` format
+        uris_match = self.gcs_uris_regex.search(ddl)
+        if not uris_match:
+            logger.warning(f"Unable to parse GCS URI from the provided DDL {ddl}.")
+            return
+
+        uris_str = uris_match.group(1)
+        source_uris = json.loads(f"[{uris_str}]")
```

Comment:
Can this fail? What should we do in that case?

Reply:
No because we already handle in above in code 
```uris_match = self.gcs_uris_regex.search(ddl)```

still we can handle this